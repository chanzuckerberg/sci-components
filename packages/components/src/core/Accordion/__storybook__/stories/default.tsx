import { Args } from "@storybook/react";
import RawAccordion, {
  AccordionDetails,
  AccordionHeader,
} from "src/core/Accordion";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Accordion = (props: Args): JSX.Element => {
  const { id, subtitle, useDivider, togglePosition } = props;
  return (
    <RawAccordion
      id={id}
      useDivider={useDivider}
      togglePosition={togglePosition}
      {...props}
    >
      <AccordionHeader id={`${id}-header`} subtitle={subtitle}>
        Accordion Header
      </AccordionHeader>
      <AccordionDetails>{EXTRA_SHORT_LOREM_IPSUM}</AccordionDetails>
    </RawAccordion>
  );
};
