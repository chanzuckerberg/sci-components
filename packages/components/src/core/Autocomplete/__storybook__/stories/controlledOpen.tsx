import { Args } from "@storybook/types";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS";
import { useState } from "react";
import RawAutocomplete from "src/core/Autocomplete";
import { AUTOCOMPLETE_WRAPPER_STYLES } from "../constants";

export const ControlledOpenDemo = <T extends DefaultAutocompleteOption>(
  props: Args
): JSX.Element => {
  const {
    label,
    multiple,
    options = AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS,
  } = props;

  const [open, setOpen] = useState(true);

  return (
    <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
      <RawAutocomplete
        onClickAway={() => {
          setOpen(false);
        }}
        onClick={() => {
          if (open) {
            setOpen(false);
          } else {
            setOpen(true);
          }
        }}
        open={open}
        id="autocomplete-demo"
        disableCloseOnSelect={multiple}
        label={label}
        multiple={multiple}
        options={options}
        getOptionDisabled={(option: T) => {
          return option.name === "Type: feature request";
        }}
        {...props}
      />
    </div>
  );
};
