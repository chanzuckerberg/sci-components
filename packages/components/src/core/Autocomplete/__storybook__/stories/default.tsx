import { Args } from "@storybook/react";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS";
import { SyntheticEvent, useEffect, useState } from "react";
import RawAutocomplete, {
  AutocompleteMultiColumnValue,
  SDSAutocompleteValue,
} from "src/core/Autocomplete";
import TagFilter from "src/core/TagFilter";
import { AutocompleteValue } from "@mui/base";
import {
  AUTOCOMPLETE_DIV_MARGIN,
  AUTOCOMPLETE_WRAPPER_STYLES,
} from "../constants";

export const Autocomplete = <
  T extends DefaultAutocompleteOption,
  Multiple extends boolean | undefined,
>(
  props: Args
): JSX.Element => {
  const {
    label,
    multiple,
    options = AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS,
    search,
    value: propValue,
    keepSearchOnSelect,
  } = props;

  type DisableClearable = false;
  type FreeSolo = false;

  const isControlled = propValue !== undefined;
  const [value, setValue] = useState<
    SDSAutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
  >(
    (multiple ? [] : null) as SDSAutocompleteValue<
      T,
      Multiple,
      DisableClearable,
      FreeSolo
    >
  );

  const [selection, setSelection] = useState<string[]>([]);

  useEffect(() => {
    if (isControlled) {
      setValue(propValue);
    }
  }, [isControlled, propValue]);

  useEffect(() => {
    setSelection([]);
    setValue(
      (multiple ? [] : null) as SDSAutocompleteValue<
        T,
        Multiple,
        DisableClearable,
        FreeSolo
      >
    );
  }, [multiple, options]);

  return (
    <>
      <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
        <RawAutocomplete
          id="autocomplete-demo"
          disableCloseOnSelect={multiple}
          label={label}
          multiple={multiple}
          onChange={handleChange}
          keepSearchOnSelect={keepSearchOnSelect}
          options={options}
          search={search}
          value={value}
          getOptionDisabled={(option: T) => {
            return option.name === "Type: feature request";
          }}
          {...props}
        />
      </div>
      <div style={{ margin: AUTOCOMPLETE_DIV_MARGIN }}>
        {selection.length
          ? selection.map((item) => {
              return (
                <TagFilter
                  key={item}
                  label={item}
                  onDelete={() => handleTagDelete(item)}
                  onClick={() => handleTagDelete(item)}
                />
              );
            })
          : null}
      </div>
    </>
  );

  // eslint-disable-next-line sonarjs/cognitive-complexity
  function handleChange(
    _: SyntheticEvent<Element, Event>,
    newValue: SDSAutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
  ) {
    setValue(newValue);
    const newSelection: string[] = [];

    if (
      (newValue as T)?.name ||
      (Array.isArray(newValue) && (newValue as T[])[0].name)
    ) {
      if (multiple) {
        setSelection(
          Array.isArray(newValue) ? newValue?.map((item) => item.name) : []
        );
      } else {
        if (newValue && !Array.isArray(newValue) && newValue.name) {
          setSelection([(newValue as T).name]);
        }
      }
    } else {
      if (multiple) {
        if (newValue) {
          Object.values(newValue).forEach((items) => {
            (items as T[])?.map(({ name }: { name: string }) => {
              newSelection.push(name);
            });
          });
          setSelection(newSelection);
        }
      } else {
        if (newValue) {
          Object.values(newValue).forEach((item) => {
            newSelection.push((item as T).name);
          });
        }
        setSelection(newSelection);
      }
    }
  }

  // eslint-disable-next-line sonarjs/cognitive-complexity
  function handleTagDeleteMultiColumn(tag: string) {
    if (multiple) {
      if (value) {
        Object.entries(value).forEach(([key, values]) => {
          const index = (values as T[])?.findIndex(
            (item: T) => item.name === tag
          );
          if (index > -1) {
            const newValue = [...(values as T[])];
            newValue.splice(index, 1);
            setValue((prev) => {
              return {
                ...prev,
                [key]: newValue as AutocompleteValue<
                  T,
                  Multiple,
                  DisableClearable,
                  FreeSolo
                >,
              };
            });

            const newSelection = [...selection];
            const deleteIndex = newSelection.indexOf(tag);
            newSelection.splice(deleteIndex, 1);
            setSelection(newSelection);
          }
        });
      }
    } else {
      if (value) {
        Object.entries(value).forEach(([key, val]) => {
          if ((val as T).name === tag) {
            const finalValue = { ...value } as AutocompleteMultiColumnValue<
              T,
              Multiple,
              DisableClearable,
              FreeSolo
            >;
            if (finalValue) delete finalValue[key];
            setValue(finalValue);

            const newSelection = [...selection];
            const deleteIndex = newSelection.indexOf(tag);
            newSelection.splice(deleteIndex, 1);
            setSelection(newSelection);
          }
        });
      }
    }
  }

  function handleTagDelete(tag: string) {
    if (
      (value && (value as T).name) ||
      (Array.isArray(value) && (value as T[])[0].name)
    ) {
      if (multiple && Array.isArray(value)) {
        const index = value?.findIndex((item) => item.name === tag);
        const newValue = [...value];
        newValue.splice(index, 1);
        setValue(
          newValue as SDSAutocompleteValue<
            T,
            Multiple,
            DisableClearable,
            FreeSolo
          >
        );

        const newSelection = [...selection];
        const deleteIndex = newSelection.indexOf(tag);
        newSelection.splice(deleteIndex, 1);
        setSelection(newSelection);
      } else {
        setValue(
          null as SDSAutocompleteValue<T, Multiple, DisableClearable, FreeSolo>
        );
        setSelection([]);
      }
    } else {
      handleTagDeleteMultiColumn(tag);
    }
  }
};
