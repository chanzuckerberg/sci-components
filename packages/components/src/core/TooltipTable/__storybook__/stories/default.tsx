import { Args } from "@storybook/react";
import Link from "src/core/Link";
import RawTooltipTableContent from "src/core/TooltipTable";

export const TooltipTableContent = (props: Args): JSX.Element => {
  const { contentAlert } = props;
  const handleAlert = () => {
    let alertContent;
    switch (contentAlert) {
      case "String":
        alertContent = "Some values do not pass the selected filter";
        break;
      case "Element":
        alertContent = <Link href="/">Click this link to see samples</Link>;
        break;
      case "None":
      default:
        alertContent = undefined;
    }
    return alertContent;
  };

  return <RawTooltipTableContent {...props} contentAlert={handleAlert()} />;
};
