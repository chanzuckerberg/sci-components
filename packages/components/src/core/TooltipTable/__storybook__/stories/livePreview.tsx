import { Args } from "@storybook/react";
import {
  TOOLTIP_TABLE_DATA,
  TOOLTIP_TABLE_LIVE_PREVIEW_STYLES,
  TOOLTIP_TABLE_TOOLTIP_STYLE_MOCK,
} from "../constants";
import Tooltip from "src/core/Tooltip";
import RawTooltipTableContent from "src/core/TooltipTable";
import { InfoOutlined } from "@mui/icons-material";

export const LivePreviewDemo = (props: Args): JSX.Element => {
  return (
    <div style={TOOLTIP_TABLE_LIVE_PREVIEW_STYLES as React.CSSProperties}>
      <div style={TOOLTIP_TABLE_TOOLTIP_STYLE_MOCK as React.CSSProperties}>
        <Tooltip
          followCursor
          placement="right-end"
          title={
            <RawTooltipTableContent {...props} data={[TOOLTIP_TABLE_DATA[0]]} />
          }
        >
          <InfoOutlined color="primary" fontSize="small" />
        </Tooltip>
      </div>

      <div style={TOOLTIP_TABLE_TOOLTIP_STYLE_MOCK as React.CSSProperties}>
        <Tooltip
          followCursor
          placement="right-end"
          title={
            <RawTooltipTableContent
              {...props}
              data={[TOOLTIP_TABLE_DATA[0], TOOLTIP_TABLE_DATA[1]]}
            />
          }
        >
          <InfoOutlined color="primary" fontSize="small" />
        </Tooltip>
      </div>

      <div style={TOOLTIP_TABLE_TOOLTIP_STYLE_MOCK as React.CSSProperties}>
        <Tooltip
          followCursor
          placement="right-end"
          title={
            <RawTooltipTableContent {...props} data={TOOLTIP_TABLE_DATA} />
          }
        >
          <InfoOutlined color="primary" fontSize="small" />
        </Tooltip>
      </div>
    </div>
  );
};
