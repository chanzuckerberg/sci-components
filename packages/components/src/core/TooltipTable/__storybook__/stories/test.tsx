import { Args } from "@storybook/react";
import RawTooltipTableContent from "src/core/TooltipTable";
import { TOOLTIP_TABLE_DATA } from "../constants";

export const TestDemo = (props: Args): JSX.Element => {
  return (
    <RawTooltipTableContent
      {...props}
      data={[TOOLTIP_TABLE_DATA[0], TOOLTIP_TABLE_DATA[1]]}
      data-testid="tooltipTable"
    />
  );
};
