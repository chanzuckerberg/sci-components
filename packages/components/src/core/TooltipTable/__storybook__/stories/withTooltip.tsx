import { Args } from "@storybook/types";
import Button from "src/core/Button";
import Link from "src/core/Link";
import TooltipCondensed from "src/core/TooltipCondensed";
import RawTooltipTableContent from "src/core/TooltipTable";

export const WithinTooltipDemo = (props: Args): JSX.Element => {
  const { contentAlert } = props;
  const handleAlert = () => {
    let alertContent;
    switch (contentAlert) {
      case "String":
        alertContent = "Some values do not pass the selected filter";
        break;
      case "Element":
        alertContent = <Link href="/">Click this link to see samples</Link>;
        break;
      case "None":
      default:
        alertContent = undefined;
    }
    return alertContent;
  };

  return (
    <div>
      Hover over the info icon to view the tooltip.
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          marginTop: "50px",
        }}
      >
        <TooltipCondensed
          title={null}
          hasInvertedStyle={false}
          componentSlot={
            <RawTooltipTableContent {...props} contentAlert={handleAlert()} />
          }
        >
          <Button
            aria-label="tooltip test button"
            sdsStyle="icon"
            sdsSize="large"
            icon="ExclamationMarkCircle"
          />
        </TooltipCondensed>
      </div>
    </div>
  );
};
