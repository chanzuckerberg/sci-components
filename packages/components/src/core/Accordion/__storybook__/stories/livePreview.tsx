import { Args } from "@storybook/react";
import { Accordion } from "./default";

const livePreviewStyles = {
  display: "grid",
  gridColumnGap: "24px",
  gridRowGap: "50px",
  gridTemplateColumns: "repeat(4, 200px)",
};

export function LivePreviewDemo(props: Args): JSX.Element {
  return (
    <div style={livePreviewStyles as React.CSSProperties}>
      <Accordion
        id="accordion-1"
        togglePosition="right"
        useDivider
        {...props}
      />
      <Accordion
        id="accordion-2"
        togglePosition="right"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
      <Accordion id="accordion-3" togglePosition="left" useDivider {...props} />
      <Accordion
        id="accordion-4"
        togglePosition="left"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
    </div>
  );
}
