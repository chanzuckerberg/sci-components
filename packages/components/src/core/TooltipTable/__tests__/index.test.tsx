import { generateSnapshots } from "@chanzuckerberg/story-utils";
import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import * as stories from "../__storybook__/index.stories";

// Returns a component that already contain all decorators from story level, meta level and global level.
const { Test } = composeStories(stories);

describe("<TooltipTable />", () => {
  generateSnapshots(stories);

  it("renders tooltip component", async () => {
    render(<Test {...Test.args} />);
    const tooltipTableElement = screen.getByTestId("tooltipTable");
    expect(tooltipTableElement).not.toBeNull();
  });

  it("renders two sections if provided two objects of data", () => {
    render(<Test {...Test.args} />);
    const sectionHeaders = screen.getAllByText(/Section/i);
    expect(sectionHeaders).toHaveLength(2);
  });

  it("renders an alert if one is passed", async () => {
    const props = {
      contentAlert: "test",
    };

    render(<Test {...Test.args} {...props} />);
    const contentAlert = screen.getByText(/test/i);
    expect(contentAlert).toBeTruthy();
  });
});
