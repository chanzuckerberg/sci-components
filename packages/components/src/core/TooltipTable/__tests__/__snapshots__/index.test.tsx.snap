// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TooltipTable /> Default story renders snapshot 1`] = `
<div
  class="MuiTableContainer-root css-1ioweka-MuiTableContainer-root"
>
  <div
    class="css-q4nnqm"
  >
    <div
      class="css-14oy9xc"
      label="Section 1"
    >
      Section 1
    </div>
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            1
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            2
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            3
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="css-q4nnqm"
  >
    <div
      class="css-14oy9xc"
      label="Section 2"
    >
      Section 2
    </div>
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 4
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            4
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 5 
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            5
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 6
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            6
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 7
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            7
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="css-q4nnqm"
  >
    <div
      class="css-14oy9xc"
      label="Section 3"
    >
      Section 3
    </div>
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 8
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            8
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 9
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            9
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1pj5u4j-MuiTableCell-root"
          >
            Label 10
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-mo86fa-MuiTableCell-root"
          >
            10
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<TooltipTable /> WithinTooltip story renders snapshot 1`] = `
<div>
  Hover over the info icon to view the tooltip.
  <div
    style="display: flex; justify-content: center; margin-top: 50px;"
  >
    <button
      aria-label="tooltip test button"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1sh3t66-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      icon="ExclamationMarkCircle"
      tabindex="0"
      type="button"
    >
      <div
        class="css-1ft6wgl"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bgw7zo-MuiSvgIcon-root"
          data-file-name="IconExclamationMarkCircleLarge"
          data-testid="IconExclamationMarkCircleLarge"
          fillcontrast="white"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        />
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
